src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -422,8 +420,8 @@ private static unsafe int SysReceiveMessageFrom(SafeSocketHandle socket, SocketF
         {
             Debug.Assert(socketAddress != null, "Expected non-null socketAddress");
 
-            int cmsgBufferLen = Interop.Sys.GetControlMessageBufferSize(isIPv4, isIPv6);
-            var cmsgBuffer = stackalloc byte[cmsgBufferLen];
+            int cmsgBufferLen = Interop.Sys.GetControlMessageBufferSize(Convert.ToInt32(isIPv4), Convert.ToInt32(isIPv6));
+            byte* cmsgBuffer = stackalloc byte[cmsgBufferLen];
```

Comment:
Unrelated question about `byte*` here. Is this being used instead of `Span<T>` because it is immediately passed along and not used in managed? I am trying to enrich my mental model for best practices and want to make sure I am getting the "why".

Reply:
It probably started as it being written before you could stackalloc into a span.  But then it hasn't changed to use span exactly for the reason you mention: we need the pointer for the interop, so if we were to stackalloc into a span, we'd need to immediately turn around to extract the pointer from it, so we might as well just get the pointer directly.  If the need for the pointer was only part of the overall operation using this, e.g. if after the interop we needed to index into the data, then the default would be to use span so that we'd get bounds checking and the like.